@W-19677915 Use einstein suggested phrases in search

sf-madhuri-uppu opened the pull request:
<img width="1521" height="742" alt="Screenshot 2025-10-10 at 11 22 11 AM" src="https://github.com/user-attachments/assets/0f7ec499-864c-456c-ae86-1843ae92abee" />
<img width="1718" height="449" alt="Screenshot 2025-10-10 at 11 22 43 AM" src="https://github.com/user-attachments/assets/65bed4bc-3a65-4eb2-8557-c88707359e9e" />

<!--- Provide a short summary of your changes in the Title field above -->

# Description

Einstein suggestion phrases in search
Suggested look - https://salesforce-internal.slack.com/archives/C0911BCGPQU/p1759868249331009


# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [x] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1)

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [x] Changes are covered by test cases
- [ ] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

sf-madhuri-uppu committed:
skip changelog

cc-prodsec commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/7695bf98-0caf-4b55-9d77-aa4595ba4e4c/license?refs=bd8ce61332e7638d1b2d575549f54425f8c90124&source=prChecksComment) |
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/7695bf98-0caf-4b55-9d77-aa4595ba4e4c?refs=bd8ce61332e7638d1b2d575549f54425f8c90124&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

sf-madhuri-uppu committed:
skip changelog

sf-kyle-wright commented on packages/template-retail-react-app/app/components/search/index.jsx:
```diff
@@ -134,7 +153,8 @@ const Search = (props) => {
                 q: searchQuery,
-                expand: 'images,prices'
+                expand: 'images,prices',
+                includeEinsteinSuggestedPhrases: true
```
@sf-madhuri-uppu can you confirm w/ search channel that this should be enabled by default? What happens if the feature is not enabled in the ecom instance - can we confirm it does not cause any issues/exceptions?

sf-madhuri-uppu replied:
@sf-kyle-wright Sure I'll check with them

sf-kyle-wright reviewed (commented):
I think overall this looks good. Please confirm the implementation stategy w/ the search team. Also my understanding is that this cannot be released until 25.10 Nov 4 GA release.

sf-madhuri-uppu committed:
skip changelog

sf-madhuri-uppu committed:
Remove unnecessary translation files

- Remove en-GB.json, en-US.json, en-XA.json from compiled translations
- Remove en-GB.json, en-US.json from source translations
- These are generated files that are not needed for Einstein suggestions functionality

sf-madhuri-uppu committed:
Generate translation files for Einstein suggestions

- Add compiled translation files (en-GB.json, en-US.json, en-XA.json)
- Add source translation files (en-GB.json, en-US.json)
- These files include translations for new Einstein suggestions features:
  - Popular Searches
  - Recent Searches
- Generated using npm run build-translations

sf-madhuri-uppu committed:
increase bundle size
